code/modules/chemistry/tools/food_and_drink.dm

```diff
@@ -1392,6 +1387,43 @@
 	initial_volume = 120
 	shard_amt = 2
 
+/obj/item/reagent_containers/food/drinks/drinkingglass/icing
+	name = "icing tube"
+	desc = "Used to put icing on cakes."
+	icon = 'icons/obj/foodNdrink/food.dmi'
+	icon_state = "icing_tube"
+	initial_volume = 50
+	amount_per_transfer_from_this = 5
+	rc_flags = RC_FULLNESS | RC_VISIBLE | RC_SPECTRO
+
+	on_reagent_change()
+		src.underlays = null
+		if (reagents.total_volume >= 0)
+			if(reagents.total_volume == 0)
+				src.icon_state = "icing_tube"
+			else
+				src.icon_state = "icing_tube_2"
+			var/datum/color/average = reagents.get_average_color()
+			var/image/chem = new /image('icons/obj/foodNdrink/food.dmi',"icing_tube_chem")
+			chem.color = average.to_rgba()
```

Comment:
if you move the image up to the object itself you could just changes its color instead of building a new image every time and then changing its color, just be sure to toss a line in disposing for it to clear the ref

Reply:
hmmm...ill look into it i see what you mean

Reply:
i did it this way so it looks beautiful and realistic. if i moved it up to the object level, parts of the icing tube that shouldnt change color, would

Reply:
beakers and such used to work like this too

Reply:
im stupid, oops, ill fix that